pkg/cmd/release-payload-controller/release_creation_status_controller.go

```diff
@@ -0,0 +1,308 @@
+// The ReleaseCreationStatusController watches for changes to the following resources:
+//   - batchv1.Jobs
+// and write the following information:
+//   - .status.releaseCreationJobResult.status
+//   - .status.releaseCreationJobResult.message
+type ReleaseCreationStatusController struct {
+	releasePayloadNamespace string
+	releasePayloadLister    releasepayloadlister.ReleasePayloadLister
+	releasePayloadClient    releasepayloadclient.ReleaseV1alpha1Interface
+
+	batchJobNamespace string
+	batchJobLister    batchv1listers.JobLister
+
+	eventRecorder events.Recorder
+
+	cachesToSync []cache.InformerSynced
+
+	queue workqueue.RateLimitingInterface
+
+	accessor meta.MetadataAccessor
+}
+
+func NewReleaseCreationStatusController(
+	releasePayloadNamespace string,
+	releasePayloadInformer releasepayloadinformer.ReleasePayloadInformer,
+	releasePayloadClient releasepayloadclient.ReleaseV1alpha1Interface,
+	batchJobNamespace string,
+	batchJobInformer batchv1informers.JobInformer,
+	eventRecorder events.Recorder,
+) (*ReleaseCreationStatusController, error) {
+	c := &ReleaseCreationStatusController{
+		releasePayloadNamespace: releasePayloadNamespace,
+		releasePayloadLister:    releasePayloadInformer.Lister(),
+		releasePayloadClient:    releasePayloadClient,
+		batchJobNamespace:       batchJobNamespace,
+		batchJobLister:          batchJobInformer.Lister(),
+		eventRecorder:           eventRecorder.WithComponentSuffix("release-creation-status-controller"),
+		queue:                   workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "ReleaseCreationStatusController"),
+	}
+
+	c.accessor = meta.NewAccessor()
+
+	c.cachesToSync = append(c.cachesToSync, releasePayloadInformer.Informer().HasSynced)
+	c.cachesToSync = append(c.cachesToSync, batchJobInformer.Informer().HasSynced)
+
+	batchJobFilter := func(obj interface{}) bool {
+		if batchJob, ok := obj.(*batchv1.Job); ok {
+			if _, ok := batchJob.Annotations[releasecontroller.ReleaseAnnotationReleaseTag]; ok {
+				return true
+			}
+		}
+		if tombstone, ok := obj.(cache.DeletedFinalStateUnknown); ok {
+			if batchJob, ok := tombstone.Obj.(*batchv1.Job); ok {
+				if _, ok = batchJob.Annotations[releasecontroller.ReleaseAnnotationReleaseTag]; ok {
+					return true
+				}
+			}
+		}
+		return false
+	}
+
+	batchJobInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
+		FilterFunc: batchJobFilter,
+		Handler: cache.ResourceEventHandlerFuncs{
+			AddFunc:    c.lookupReleasePayload,
+			UpdateFunc: func(old, new interface{}) { c.lookupReleasePayload(new) },
+			DeleteFunc: c.lookupReleasePayload,
+		},
+	})
+
+	// In case someone/something deletes the ReleaseCreationJobResult.Status, try and rectify it...
+	releasePayloadFilter := func(obj interface{}) bool {
+		if releasePayload, ok := obj.(*v1alpha1.ReleasePayload); ok {
+			if len(releasePayload.Status.ReleaseCreationJobResult.Status) == 0 {
```

Comment:
if the message is missing, you should also return true.

Reply:
Done